docs/build.md

```diff
@@ -0,0 +1,226 @@
+# Build
+
+- [Overview](#overview)
+- [Build Controller](#build-controller)
+- [Configuring a Build](#configuring-a-build)
+  - [Defining the Source](#defining-the-source)
+  - [Defining the Strategy](#defining-the-strategy)
+  - [Defining the Builder or Dockerfile](#defining-the-builder-or-dockerfile)
+  - [Defining Resources](#defining-resources)
+  - [Defining the Output](#defining-the-output)
+
+## Overview
+
+A `Build` resource allows the user to define:
+
+- source
+- strategy
+- builder
+- dockerfile
+- resources
+- output
+
+A `Build` is available within a namespace.
+
+## Build Controller
+
+The controller watches for:
+
+- Updates on the `Build` resource (_CRD instance_)
+
+When the controller reconciles it:
+
+- Validates if the referenced `StrategyRef` exists.
+- Validates if the container `registry` output secret exists.
+
+## Configuring a Build
+
+The `Build` definition supports the following fields:
+
+- Required:
+  - [`apiVersion`](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#required-fields) - Specifies the API version, for example `build.dev/v1alpha1`.
+  - [`kind`](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#required-fields) - Specifies the Kind type, for example `Build`.
+  - [`metadata`](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#required-fields) - Metadata that identify the CRD instance, for example the name of the `Build`.
+  - `spec.source.URL` - Refers to the Git repository containing the source code.
+  - `spec.strategy` - Refers to the `BuildStrategy` to be used, see the [examples](../samples/buildstrategy)
+  - `spec.builder` - Refers to the image containing the build tools. (_when used, `spec.dockerfile` should not be used_)
+  - `spec.dockerfile` - Path to a Dockerfile to be used for building an image. (_when used, `spec.builder` should not be used_)
+  - `spec.output`- Refers to the location where the generated image would be pushed.
+  - `spec.output.credentials.name`- Reference an existing secret to get access to the container registry.
+
+- Optional:
+  - `spec.source.Revision` - Refers to the a git specific branch, default is master.
```

Comment:
This is somehow duplicated because the source is described below? I suggest to move the other source properties up to here (contextDir).

Reply:
**Optional** hosts only the paths that the user can avoid setting. I `spec.source.Revision` is not set, then it will default to `master`. `spec.source.URL` does not belongs to `Optional` because users are expected to set from where the source code is coming. 
Also, they are not duplicates because the paths are differently. Do you agree?